packages/react-dom/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -1897,5 +1952,70 @@ export function writeClientRenderBoundaryInstruction(
   }
 
   writeChunk(destination, boundaryID);
+  if (errorHash || errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorHash)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorHash)),
+      );
+  }
+  if (errorMessage || errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    if (errorMessage)
+      writeChunk(
+        destination,
+        stringToChunk(escapeJSStringsForInstructionScripts(errorMessage)),
+      );
+  }
+  if (errorComponentStack) {
+    writeChunk(destination, clientRenderErrorScriptArgInterstitial);
+    writeChunk(
+      destination,
+      stringToChunk(escapeJSStringsForInstructionScripts(errorComponentStack)),
+    );
+  }
   return writeChunkAndReturn(destination, clientRenderScript2);
 }
+
+const regexForJSStringsInScripts = /[<\'\`\u2028\u2029]/g;
+function escapeJSStringsForInstructionScripts(input: string): string {
+  if (typeof input !== 'string') {
+    // eslint-disable-next-line react-internal/prod-error-codes
+    throw new Error(
+      `regexForJSStringsInScripts must be passed a string but was passed something of type "${
+        input === null ? 'null' : typeof input
+      }" instead`,
+    );
+  }
+  let escaped = JSON.stringify(input);
+  escaped = escaped.replace(regexForJSStringsInScripts, match => {
+    switch (match) {
+      // santizing breaking out of strings and script tags
+      case '<':
+        return '\\u003c';
+      case "'":
```

Comment:
strictly speaking it isn't since we control the wrapping quotations. but if we did a careless quote replacement for our strings and ended up with single quotes for attributes then this would allow escaping. I put it here so that even if we change later the quotes used we won't have to remember to modify the escape procedure

Reply:
actually I'm wrong. I think I was mixing up the html attribute context with the js argument one when i was deciding to include these